internal/store/store.go

```diff
@@ -461,6 +461,24 @@ func (s *Store) PushCompose(composeID uuid.UUID, bp *blueprint.Blueprint, checks
 	return nil
 }
 
+func (s *Store) CancelCompose(id uuid.UUID) error {
+	return s.change(func() error {
+		compose, exists := s.Composes[id]
+
+		if !exists {
+			return &NotFoundError{}
+		}
+
+		if compose.QueueStatus != "WAITING" && compose.QueueStatus != "RUNNING" {
+			return &InvalidRequestError{fmt.Sprintf("Compose %s is not in WAITING or RUNNING.", id)}
+		}
+
+		delete(s.Composes, id)
```

Comment:
After many layers of abstraction, this will crash the worker when it tries to update the status of non-existing compose. However, it will be restarted by systemd. From osbuild-worker@.service:
```
Restart=on-failure
RestartSec=10s
```

I just wanna double check whether we are OK with depending on systemd restarting our stuff when user cancels a job. It surely works but looks a bit hacky to me.

Reply:
Hmmmm..... the intention was that it should restart, but not crash. I'll fix this.